code/kopilua/src/lvm.cs

```diff
@@ -354,10 +354,10 @@ public static void luaV_concat( lua_State L, int total, int last )
 					for ( i = n; i > 0; i-- )
 					{  /* concat all strings */
 						uint l = tsvalue( top - i ).len;
-						memcpy( buffer.chars, (int)tl, svalue( top - i ).chars, (int)l );
+						svalue( top - i ).CopyTo( 0, buffer.chars, (int)tl, (int)l );
 						tl += l;
 					}
-					setsvalue2s( L, top - n, luaS_newlstr( L, buffer, tl ) );
+					setsvalue2s( L, top - n, luaS_newstr( L, buffer.ToString() ) );
```

Comment:
Make sure t1 is the actual buffer length. Might need to actually do a new string()